src/integration-test/java/fi/hsl/transitdata/tripupdate/application/ITTripUpdateProcessor.java

```diff
@@ -185,6 +186,85 @@ private void validateCancellationPayload(final GtfsRealtime.FeedMessage feedMess
+
+                    logger.info("Cancellation sent, reading it back");
+
+                    Message<byte[]> received = TestPipeline.readOutputMessage(context);
+                    assertNotNull(received);
+                    validatePulsarProperties(received, Long.toString(dvjId), ts, TransitdataProperties.ProtobufSchema.GTFS_TripUpdate);
+                    validateScheduledRelationshipAndStopEstimateCountFromGtfsRtFeedMessage(received.getData(),
+                            GtfsRealtime.TripDescriptor.ScheduleRelationship.CANCELED, 0);
+                }
+                {
+                    final long ts = System.currentTimeMillis();
+                    sendPubtransSourcePulsarMessage(context.source, new PubtransPulsarMessageData.StopEstimateMessageData(estimate, ts, dvjId));
+
+                    logger.info("Estimate sent, reading it back");
+
+                    Message<byte[]> received = TestPipeline.readOutputMessage(context);
+                    // Trip should be cancelled so we shouldn't receive an estimate.
+                    assertNull(received);
+                }
+                {
+                    final long ts = System.currentTimeMillis();
+                    sendPubtransSourcePulsarMessage(context.source, new PubtransPulsarMessageData.CancellationPulsarMessageData(running, ts, dvjId));
+
+                    logger.info("Running sent, reading it back");
+
+                    Message<byte[]> received = TestPipeline.readOutputMessage(context);
+                    // Now we should receive the estimate with correct status and it should include the estimate sent earlier.
+                    assertNotNull(received);
+                    validatePulsarProperties(received, Long.toString(dvjId), ts, TransitdataProperties.ProtobufSchema.GTFS_TripUpdate);
+                    validateScheduledRelationshipAndStopEstimateCountFromGtfsRtFeedMessage(received.getData(),
+                            GtfsRealtime.TripDescriptor.ScheduleRelationship.SCHEDULED, 1);
+
+                }
+                {
+                    final long ts = System.currentTimeMillis();
+                    sendPubtransSourcePulsarMessage(context.source, new PubtransPulsarMessageData.CancellationPulsarMessageData(cancellation, ts, dvjId));
+
+                    logger.info("Cancelling again, reading it back");
+
+                    Message<byte[]> received = TestPipeline.readOutputMessage(context);
+                    assertNotNull(received);
+                    validatePulsarProperties(received, Long.toString(dvjId), ts, TransitdataProperties.ProtobufSchema.GTFS_TripUpdate);
+                    //Cancellation should strip out the estimates again.
+                    validateScheduledRelationshipAndStopEstimateCountFromGtfsRtFeedMessage(received.getData(),
+                            GtfsRealtime.TripDescriptor.ScheduleRelationship.CANCELED, 0);
```

Comment:
Ok so if `ScheduledRelationship` equals `CANCELED`, a `FeedMessage` with an empty list is sent?

Reply:
yep, that's how we've planned it with @laaksma. We need to keep sending the cancellations but no need to add any estimates for it, since it's cancelled.